test/unit/path_spec.lua

```diff
@@ -603,4 +603,21 @@ describe('path.c', function()
       eq(FAIL, path_is_absolute('not/in/my/home~/directory'))
     end)
   end)
+
+  describe('path_with_extension', function()
+    local function path_with_extension(filename, extension)
+      local c_filename = to_cstr(filename)
+      local c_extension = to_cstr(extension)
+      return cimp.path_with_extension(c_filename, c_extension)
+    end
```

Comment:
this helper function is not needed actually because of using `const char*`(according to http://luajit.org/ext_ffi_semantics.html#convert_fromlua) but I used the example from `path_is_absolute`